pinot-query-planner/src/main/java/org/apache/pinot/query/planner/plannode/JoinNode.java

```diff
@@ -94,24 +92,24 @@ public <T, C> T visit(PlanNodeVisitor<T, C> visitor, C context) {
 
   public static class JoinKeys {
     @ProtoProperties
-    private KeySelector<Object[], Object[]> _leftJoinKeySelector;
+    private List<Integer> _leftKeys;
     @ProtoProperties
-    private KeySelector<Object[], Object[]> _rightJoinKeySelector;
+    private List<Integer> _rightKeys;
```

Comment:
superb refactoring. this was a legacy property we added for pluggability that we never used. we should remove these and use factory methods!